src/interpreter/mod.rs

```diff
@@ -329,20 +350,85 @@ fn parse_statement(
             drop(variables);
             let var = function.create_variable();
             let expr = &token.children[1];
-            let expr = parse_expr(expr, writer, scope)?;
+            let expr = parse_expr(expr, engine, scope)?;
             scope
                 .variables
                 .borrow_mut()
                 .insert(name.to_string(), VariableType::Stack(var));
             Expression::AssignStack(var, expr.into())
         }
+        "structDeclaration" => parse_struct_declaration(token, engine, scope, function)?,
         name => unreachable!("{name}"),
     })
 }
 
+fn parse_struct_declaration(
+    token: &Token,
+    engine: &mut ExecutionEngine<FenderTypeSystem>,
+    scope: &mut LexicalScope,
+    outer_function: &mut FunctionWriter<FenderTypeSystem>,
+) -> InterpreterResult {
+    let mut struct_name = String::new();
+    let mut fields = Vec::new();
+
+    for child_token in token.iter() {
+        match child_token.get_name().as_deref().unwrap() {
+            "name" => struct_name = child_token.get_match(),
+            "structBody" => {
+                for arg_token in child_token.iter() {
+                    fields.push((
```

Comment:
Too much is being passed to this function, this should be split up and use temporary variables